sys-block/gpart/patches/gpart-0.3.patchset

```diff
@@ -0,0 +1,5672 @@
++g_module *g_mod_lookup(int how, char *name)
+ {
+-	g_module	*m;
++	g_module *m;
+ 
+-	if (g_head == 0)
+-	{
++	if (g_head == 0) {
+ 		if (how == GM_LOOKUP)
+ 			return (0);
+ 		g_head = (g_module *)alloc(sizeof(g_module));
+ 		m = g_head;
+-	}
+-	else
+-	{
++	} else {
+ 		for (m = g_head; m->m_next; m = m->m_next)
+-			if (strcmp(m->m_name,name) == 0)
++			if (strcmp(m->m_name, name) == 0)
+ 				return (m);
+ 		if (how == GM_LOOKUP)
+ 			return (0);
+@@ -125,8 +101,9 @@ g_module *g_mod_lookup(int how,char *name)
+ 		m = m->m_next;
+ 	}
+ 	if ((m->m_name = strdup(name)) == 0)
+-		pr(FATAL,"out of memory in strdup");
+-	m->m_weight = 1.0; g_count++;
++		pr(FATAL, "out of memory in strdup");
++	m->m_weight = 1.0;
++	g_count++;
+ 	return (m);
+ }
+ 
+@@ -148,10 +125,10 @@ void g_mod_addinternals()
+ 		} \
+ 	} while (0);
+ 
+-	/*
+-	 * If no weights are given on the command line, the order
+-	 * is somehow important.
+-	 */
++/*
++ * If no weights are given on the command line, the order
++ * is somehow important.
++ */
+ 
+ #define G_MODULE(mod) GMODINS(mod)
+ G_MODULES
+diff --git a/src/gpart.c b/src/gpart.c
+index 8555f66..936ceb0 100644
+--- a/src/gpart.c
++++ b/src/gpart.c
+@@ -13,7 +13,7 @@
+  * Modified:  11.06.1999 <mb@ichabod.han.de>
+  *            Handle disk read errors.
+  *            Minor fixes.
+- * 
++ *
+  *            29.06.1999 <mb@ichabod.han.de>
+  *            Made every disk read/write buffer aligned to pagesize.
+  *
+@@ -21,7 +21,7 @@
+  *            Default scan increment now 's'.
+  *            Extended ptbl boundary condition now depends on scan
+  *            increment.
+- * 
++ *
+  *            26.02.2000 <mb@ichabod.han.de>
+  *            Default scan increment 'h' again.
+  *            Fixed faulty head boundary condition.
+@@ -45,7 +45,6 @@
+  *
+  */
+ 
+-
+ #include <stdio.h>
+ #include <stdlib.h>
+ #include <stdarg.h>
+@@ -58,317 +57,304 @@
+ #include <sys/types.h>
+ #include "gpart.h"
+ 
++static const char *gpart_version = PACKAGE_NAME " v" VERSION;
+ 
+-static const char	*gpart_version = PACKAGE_NAME " v" VERSION;
+-
+-
+-int		f_check = 0, f_verbose = 0, f_dontguess = 0, f_fast = 1;
+-int		f_getgeom = 1, f_interactive = 0, f_quiet = 0, f_testext = 1;
+-int		f_skiperrors = 1, berrno = 0;
+-int		(*boundary_fun)(disk_desc *,s64_t);
+-unsigned long	increment = 's', gc = 0, gh = 0, gs = 0;
+-s64_t		skipsec = 0, maxsec = 0;
+-FILE		*logfile = 0;
+-
+-
++int f_check = 0, f_verbose = 0, f_dontguess = 0, f_fast = 1;
++int f_getgeom = 1, f_interactive = 0, f_quiet = 0, f_testext = 1;
++int f_skiperrors = 1, berrno = 0;
++int (*boundary_fun)(disk_desc *, s64_t);
++unsigned long increment = 's', gc = 0, gh = 0, gs = 0;
++s64_t skipsec = 0, maxsec = 0;
++FILE *logfile = 0;
+ 
+ void usage()
+ {
+-	FILE		*fp = stderr;
+-
+-	fprintf(fp,"Usage: %s [options] device\n",PACKAGE_NAME);
+-	fprintf(fp,"Options: [-b <backup MBR>][-C c,h,s][-c][-d][-E][-e][-f][-g][-h][-i]\n");
+-	fprintf(
```

Comment:
We have these in [`<endian.h>`](http://xref.plausible.coop/source/xref/haiku/headers/compatibility/bsd/endian.h#33). Define `_BSD_SOURCE` to use them